packages/entities/entities-shared/src/components/entity-link/EntityLink.vue

```diff
@@ -0,0 +1,149 @@
+<template>
+  <div
+    class="kong-ui-public-entity-link"
+    data-testid="entity-link-parent"
+  >
+    <div
+      v-if="isDeletedEntity"
+      class="deleted-entity"
+    >
+      {{ deletedUuidDisplay }}
+    </div>
+    <div v-else-if="!entityUuid">
+      {{ ' – ' }}
+    </div>
+    <KExternalLink
+      v-else-if="entityUuid"
+      class="entity-link"
+      :href="externalLink"
+      target="_blank"
+    >
+      <KTooltip
+        :label="isTruncated && entityLinkData.label || ''"
+      >
+        <span
+          ref="textContent"
+          class="entity-link-label"
+        >
+          {{ entityLinkData.label }}
+        </span>
+      </KTooltip>
+    </KExternalLink>
+    <KTooltip
+      v-if="entityUuid"
+      :key="copyUuidTooltipText"
+      class="copy-uuid-tooltip"
+      :label="copyUuidTooltipText"
+      max-width="160"
+      placement="bottomEnd"
+    >
+      <KClipboardProvider v-slot="{ copyToClipboard }">
+        <span
+          class="entity-link-copy-id"
+          @click.stop="onCopyUuid(copyToClipboard)"
+        >
+          <CopyIcon
+            class="copy-icon"
+            :color="KUI_COLOR_TEXT_PRIMARY"
+            data-testid="copy-id"
+            :size="16"
+          />
+        </span>
+      </KClipboardProvider>
+    </KTooltip>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, ref } from 'vue'
+import type { PropType, Ref } from 'vue'
+import type { EntityLinkData } from '../../types'
+import composables from '../../composables'
+import { CopyIcon } from '@kong/icons'
+import { KUI_COLOR_TEXT_PRIMARY } from '@kong/design-tokens'
+
+const props = defineProps({
+  entityLinkData: {
+    type: Object as PropType<EntityLinkData>,
+    required: true,
+  },
+  externalLink: {
+    type: String,
+    required: false,
+    default: '',
+  },
+})
+
+const textContent = ref<HTMLElement>()
+const { isTruncated } = composables.useTruncationDetector(textContent as Ref<HTMLElement>)
+const { i18n: { t } } = composables.useI18n()
+
+const tooltipDefaultText = t('global.actions.copyId')
+const copyUuidTooltipText = ref(tooltipDefaultText)
+
+// If the dimension value is the same as the label, it means the entity has been deleted
+const isDeletedEntity = computed(() => {
+  return props.entityLinkData.deleted
```

Comment:
```suggestion
  return props.entityLinkData.deleted ?? false
```

… if we make `deleted` optional